tests/test_diffraction_objects.py

```diff
@@ -341,5 +342,36 @@ def test_dump(tmp_path, mocker):
 
 @pytest.mark.parametrize("inputs, expected", tc_params)
 def test_constructor(inputs, expected):
-    actualdo = DiffractionObject(**inputs)
-    compare_dicts(actualdo.__dict__, expected)
```

Comment:
Can we use a library `DeepDiff` to compare two dicts instead? We've manually defined `compare_dicts` and `dicts_equal` within `test_diffraction_objects.py` which can be replaced using `DeepDiff` across all .py files.